filters/special_casing/filter.py

```diff
@@ -0,0 +1,20 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class TextSpecialCasingFilter(SentenceOperation):
+    """Filter if any of the special casings appear in the sentence:
+    All text is capitalized or All text is lowercased or there is Title Casing in the Sentence.
+    """
+
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+
+    def filter(self, sentence: str = None) -> bool:
+
+        return any(
+            [sentence.isupper(), sentence.islower(), sentence.istitle()]
```

Comment:
Just as a sanity check, I would suggest to double-check that no such filter has been proposed before.

Reply:
Is it important for us to know of which type was the sentence?
My understanding that as of know the filter only tells that the sentence is one of three types but does not tell which one exactly.

Reply:
Thanks, I searched the PRs using the 'casing' keyword, it seems there isn't any casing based filter. The filter picks up sentences that are weird in terms of casing (vs normal "text"), so it's a single category in itself. We can return the type too, but does the api allow the return of explicit values for filters? Wouldn't that mean that it is better to have three filters here, one for each casing type so that the user can apply filters compositionally without any post-processing from their side? If we return explicit values, then aren't we basically asking the user to do further filtering themselves? Maybe @kaustubhdhole could weight in. Thanks.

Reply:
> it seems there isn't any casing based filter.

I agree, looks like there no casing based filters.

Reply:
> The filter picks up sentences that are weird in terms of casing (vs normal "text"), so it's a single category in itself.

Okay, I buy the argument that all these cases can be considered all together as "weird".

Reply:
> Wouldn't that mean that it is better to have three filters here, one for each casing type so that the users could apply filters compositionally without any post-processing from their side?

Fair point. The idea about three simple separate filters might still be valid though.